File: scripts/insert_ga.py
import os
import re

GA_TAG = """<script async src="https://www.googletagmanager.com/gtag/js?id=G-39MBNBLCY1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-39MBNBLCY1');
</script>"""

MEASUREMENT_ID = "G-39MBNBLCY1"

def insert_ga_tag(file_path):
    try:
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()
    except Exception as e:
        print(f"Could not read {file_path}: {e}")
        return False

    # Check if already inserted
    if MEASUREMENT_ID in content:
        print(f"Skipped (already contains ID): {file_path}")
        return False

    # Search for <head> tag
    match = re.search(r'(<head\b[^>]*>)', content, re.IGNORECASE)
    if not match:
        print(f"Skipped (no <head> tag found): {file_path}")
        return False

    head_tag = match.group(1)
    # Insert GA_TAG right after <head> tag
    new_content = content.replace(head_tag, f"{head_tag}\n{GA_TAG}")

    try:
        with open(file_path, 'w', encoding='utf-8') as f:
            f.write(new_content)
        print(f"Successfully injected: {file_path}")
        return True
    except Exception as e:
        print(f"Could not write {file_path}: {e}")
        return False

def main():
    # Target directory is the workspace root (parent of the scripts folder)
    workspace_dir = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
    print(f"Scanning workspace: {workspace_dir}")
    
    html_files = []
    exclude_dirs = {'.git', 'node_modules', '.next'}
    
    for root, dirs, files in os.walk(workspace_dir):
        # Prune excluded directories
        dirs[:] = [d for d in dirs if d not in exclude_dirs]
        
        for file in files:
            if file.endswith('.html'):
                html_files.append(os.path.join(root, file))

    print(f"Found {len(html_files)} HTML files.")
    modified_count = 0
    for file_path in html_files:
        if insert_ga_tag(file_path):
            modified_count += 1
            
    print(f"Done. Modified {modified_count} out of {len(html_files)} files.")

if __name__ == '__main__':
    main()

File: scripts/insert_ga.js
const fs = require('fs');
const path = require('path');

const GA_TAG = `<script async src="https://www.googletagmanager.com/gtag/js?id=G-39MBNBLCY1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-39MBNBLCY1');
</script>`;

const MEASUREMENT_ID = 'G-39MBNBLCY1';

function insertGaTag(filePath) {
  try {
    let content = fs.readFileSync(filePath, 'utf8');

    // Check if already inserted
    if (content.includes(MEASUREMENT_ID)) {
      console.log(`Skipped (already contains ID): ${filePath}`);
      return false;
    }

    // Search for <head> tag
    const match = content.match(/(<head\b[^>]*>)/i);
    if (!match) {
      console.log(`Skipped (no <head> tag found): ${filePath}`);
      return false;
    }

    const headTag = match[1];
    // Insert GA_TAG right after <head> tag
    const newContent = content.replace(headTag, `${headTag}\n${GA_TAG}`);

    fs.writeFileSync(filePath, newContent, 'utf8');
    console.log(`Successfully injected: ${filePath}`);
    return true;
  } catch (error) {
    console.error(`Error processing ${filePath}:`, error.message);
    return false;
  }
}

function walkDir(dir, callback) {
  const files = fs.readdirSync(dir);
  for (const file of files) {
    const filePath = path.join(dir, file);
    const stat = fs.statSync(filePath);
    if (stat.isDirectory()) {
      if (file !== 'node_modules' && file !== '.git' && file !== '.next') {
        walkDir(filePath, callback);
      }
    } else if (file.endsWith('.html')) {
      callback(filePath);
    }
  }
}

function main() {
  const workspaceDir = path.dirname(__dirname);
  console.log(`Scanning workspace: ${workspaceDir}`);

  let totalFiles = 0;
  let modifiedCount = 0;

  walkDir(workspaceDir, (filePath) => {
    totalFiles++;
    if (insertGaTag(filePath)) {
      modifiedCount++;
    }
  });

  console.log(`Done. Modified ${modifiedCount} out of ${totalFiles} files.`);
}

main();
